metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -551,23 +569,50 @@ def _process_schema(
             yield from self.gen_schema_containers(snowflake_schema, db_name)
 
         if self.config.include_tables:
-            snowflake_schema.tables = self.get_tables_for_schema(
-                conn, schema_name, db_name
-            )
+            try:
+                snowflake_schema.tables = self.get_tables_for_schema(
+                    conn, schema_name, db_name
+                )
+            except SnowflakePermissionError as e:
+                # Ideal implementation would use PEP 678 – Enriching Exceptions with Notes
+                error_msg = f"Failed to list tables for schema {db_name}.{schema_name}. Please check permissions."
+                raise SnowflakePermissionError(error_msg) from e.__cause__
```

Comment:
same question as above, and there's one more below too

Reply:
addressed above.